resources/js/common.js

```diff
@@ -283,3 +282,40 @@ function postwith (to,p) {
   document.body.removeChild(myForm) ;
 }
 
+
+function isValidDate(dateString)

```

Comment:
After sync'd  with the master branch and  conflict resolving done, this PR can be tested again and ready to merge.  The following changes have been made to the original PR.
1. Date validation dropped.
2. Extra cost fields in the  Cost column in "exported" CSV dropped. If later better idea can be acquired about how to present all cost fields in the  Cost column, a new PR can be created.